code/modules/jobs/job_types/_job.dm

```diff
@@ -133,6 +155,112 @@
+/// Returns the number of players who are inside a role if the job hasn't reached
+/// its population limit, otherwise groups together roles by checking for their
+/// min_pop_redirect proxy role.
+/datum/job/proc/get_group_position_count()
+	var/spawn_group_size = current_positions * dynamic_spawn_group_multiplier
+	// Find all jobs that proxy to the target's spawn group
+	// This will mean that medical will count all of the players in medical
+	for (var/datum/job/group_job in SSjob.occupations)
+		// We already counted ourselves
+		if (group_job == src)
+			continue
+		// If this job proxies to us, then their proxy needs to be active
+		if (group_job.min_pop_redirect == type && SSjob.initial_players_to_assign < group_job.min_pop)
+			// If the HOP adds a geneticist slot, and that slot gets taken then it counts as if
+			// there is no geneticist at all.
+			// If the HOP adds a geneticist slot and it doesn't get taken, then it has no effect.
+			spawn_group_size += max(0, group_job.current_positions * group_job.dynamic_spawn_group_multiplier - group_job.total_position_delta)
+		// If we proxy to this job, then our proxy needs to be active
+		if (min_pop_redirect == group_job.type && SSjob.initial_players_to_assign < min_pop)
+			spawn_group_size += max(0, group_job.current_positions * group_job.dynamic_spawn_group_multiplier - group_job.total_position_delta)
+		// If we are sharing a proxy, both proxies need to be active
+		if (min_pop_redirect != null && min_pop_redirect == group_job.min_pop_redirect && SSjob.initial_players_to_assign < group_job.min_pop && SSjob.initial_players_to_assign < min_pop)
+			spawn_group_size += max(0, group_job.current_positions * group_job.dynamic_spawn_group_multiplier - group_job.total_position_delta)
+	return spawn_group_size - total_position_delta
+
+/// Get the number of positions that are available for this job.
+/// Will typically return the total_positions value with the delta set by the HOP added on.
+/// If the min/max pop is not met returns 0.
+/// If the total positions is -1, returns -1 representing an unlimited position count
+/// If a dynamic spawn group is set, or the min pop is not met and the dynamic spawn group of the min_pop_redirect
+/// role is set, then it will compute the number of positions available based off of how populated the other jobs
+/// in the group are.
+/// Certain roles such as geneticist will return the smallest the smaller value between the number of jobs remaining
+/// in the medical department, and the max limit of geneticist slots available, if the lowpop limit has not been met.
+/// = Params =
+/// ignore_self_limit: bool => If set, then the max limit of the job will be ignored when using the dynamic group
+/// scaling calculations.
+/datum/job/proc/get_spawn_position_count(ignore_self_limit = FALSE)
+	var/player_count = SSjob.initial_players_to_assign
+	// SSjob has not been allocated yet
+	if (!player_count)
+		player_count = length(GLOB.clients)
```

Comment:
I think this is wrong condition.
Let's say we have 10 clients, and a round has been started, while nobody is ready.
`player_count` should be 0, but `length(GLOB.clients)` will be 10.
I think this should just early return?
If that's intended, I recommend writing a comment.

Reply:
I kind of agree, if it hasn't been assigned then we shouldn't be calling this proc but I won't be making changes for two reasons:
- Changing it to crash if we have null player count assigned would require retesting everything
- This proc may be used in a niche place that I don't know about, which requires some assumptions to be made for the sake of correctly showing some data